code/datums/status_effects/buffs.dm

```diff
@@ -393,3 +393,40 @@
 			owner.emote("gasp")
 	cling.genetic_damage += stacks
 	cling = null
+
+/datum/status_effect/chainsaw_slaying
+	id = "chainsaw_slaying"
+	duration = 5 SECONDS
+	status_type = STATUS_EFFECT_REFRESH
+	alert_type = /obj/screen/alert/status_effect/chainsaw
+
+/obj/screen/alert/status_effect/chainsaw
+	name = "Revved up!"
+	desc = "<span class='danger'>... guts, huge guts! Kill them... must kill them all!</span>"
+	icon_state = "chainsaw"
+
+/datum/status_effect/chainsaw_slaying/on_apply()
+	. = ..()
+	if(.)
+		if(ishuman(owner))
+			var/mob/living/carbon/human/H = owner
+			H.physiology.brute_mod *= 0.8
+			H.physiology.burn_mod *= 0.8
+			H.physiology.stamina_mod *= 0.8
+		add_attack_logs(owner, owner, "gained chainsaw stun immunity", ATKLOG_ALL)
+		owner.add_stun_absorption("chainsaw", INFINITY, 4)
+		owner.playsound_local(get_turf(owner), 'sound/effects/singlebeat.ogg', 40, TRUE, use_reverb = FALSE)
+
+/datum/status_effect/chainsaw_slaying/on_remove()
+	add_attack_logs(owner, owner, "lost chainsaw stun immunity", ATKLOG_ALL)
+	if(islist(owner.stun_absorption) && owner.stun_absorption["chainsaw"])
+		owner.stun_absorption -= "chainsaw"
+	if(ishuman(owner))
+		var/mob/living/carbon/human/H = owner
+		H.physiology.brute_mod /= 0.8
+		H.physiology.burn_mod /=0.8
+		H.physiology.stamina_mod /= 0.8
+		var/stamina = H.getStaminaLoss()
+		if(stamina < 100)
+			return
+		H.enter_stamcrit()
```

Comment:
this is kinda weird. I get why you need to do it, however this still leaves the bug for the other implementations of stun absorption.
might want to make a general `remove_stun_aborption` which does this.